src/pybind/mgr/cephadm/migrations.py

```diff
@@ -0,0 +1,116 @@
+import logging
+from typing import List, TYPE_CHECKING, Iterator
+
+from ceph.deployment.service_spec import PlacementSpec, ServiceSpec, HostPlacementSpec
+from cephadm.inventory import Inventory, SpecStore, HostCache
+from cephadm.schedule import HostAssignment
+
+from orchestrator import DaemonDescription, OrchestratorError
+
+if TYPE_CHECKING:
+    from .module import CephadmOrchestrator
+
+LAST_MIGRATION = 1
+
+logger = logging.getLogger(__name__)
+
+
+
+class Migrations:
+    def __init__(self, mgr: "CephadmOrchestrator"):
+        self.mgr = mgr
+
+        self.initialize()
+
+    def initialize(self):
+        if self.mgr.migration_current is None:
+            self.set(0)
+        self.migrate()
+
+    def set(self, val):
+        self.mgr.set_module_option('migration_current', val)
+        self.mgr.migration_current = val
+
+    def is_migration_ongoing(self):
+        return self.mgr.migration_current != LAST_MIGRATION
+
+    def verify_no_migration(self):
+        if self.is_migration_ongoing():
+            raise OrchestratorError("cephadm migration still ongoing. Please wait, until the migration is complete.")
```

Comment:
instead of raising, shouldn't we maybe write to the `cluster_health_dict` (\<insert correct reference\>) since this is being called in the main loop

Reply:
I think we should add a general exception handler in `serve()` anyway.

Reply:
yep, I think _that's_ the right approach.. we can keep on raising exception across the codebase and decide in `serve()` which exceptions are worthy to _actually_ raise and which should be just warnings.

Reply:
but when do we want to clear the warning again?

Reply:
what happens when this exception is raised ... is it only visible from the mgr log? 

I somewhat like the idea of seeing some kind of status/progress via cluster health  ..

Reply:
> but when do we want to clear the warning again?

I'd assume that this codepath is exercised quite a bit and we have to react to the case where `is_migration_ongoing` is `False` by unsetting the flag.

Reply:
I'd like to leave this outside of this PR, if possible.